hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/KeyDeletingService.java

```diff
@@ -88,6 +93,9 @@ public class KeyDeletingService extends AbstractKeyDeletingService {
   private final int keyLimitPerTask;
   private final AtomicLong deletedKeyCount;
   private final AtomicBoolean suspended;
+  private final Map<String, Long> exclusiveSizeList;
+  private final Map<String, Long> exclusiveReplicatedSizeList;
+  private final Set<String> completedExclusiveSizeList;
```

Comment:
nit: I think this could be a local variable. I see you are clearing it after updating the snapshot info with exclusive size but someone might remove the line by mistake and no test will catch it unless it is explicitly checking `completedExclusiveSizeList` after the run.

Reply:
The reason for keeping it global is because of the `keyLimitPerTask`. At a time it is not guaranteed the entire snapshot is processed, If we keep it local it will be cleared in the next iteration. I will be doing a follow-up PR, where we will be continuing size calculation from where we stopped during the pervious iteration because of `keyLimitPerTask`